hw/ip/prim/doc/prim_flash.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Primitive Component: Flash Wrapper"
+---
+
+# Overview
+`prim_flash` is a wrapper interface for technology specific flash modules.
+
+As the exact details of each technology can be different, this document mainly describes the interface requirements and their functions.
+The wrapper however does assume that all page sizes are the same (they cannot be different between data and info partitions, or different types of info partitions).
+
+The
+
+## Parameters
+
+Name           | type   | Description
+---------------|--------|----------------------------------------------------------
+NumBanks       | int    | Number of flash banks.  Flash banks are assumed to be identical, asymmetric flash banks are not supported
+InfosPerBank   | int    | Maximum number of info pages in the info partition.  Since info partitions can have multiple types, this is max among all types.
+InfoTypes      | int    | The number of info partition types, this number can be 1~N.
+InfoTypesWidth | int    | The number of bits needed to represent the info types.
+PagesPerBank   | int    | The number of pages per bank for data partition.
+WordsPerPage   | int    | The nubmer of words per page per bank for both information and data partition.
+DataWidth      | int    | The full data width of a flash word (inclusive of metadata)
+MetaDataWidth  | int    | The metadata width of a flash word
+TestModeWidth  | int    | The number of test modes for a bank of flash
+
+
+## Signal Interfaces
+
+### Overall Interface Signals
+Name                 | In/Out | Description
+---------------------|--------|---------------------------------
+clk                  | input  | Clock input
+rst_n                | input  | Reset input
+flash_req            | input  | Inputs from flash protocol and physical controllers
+flash_rsp            | output | Outputs to flash protocol and physical controllers
+prog_type_avail      | output | Avaialble program types in this flash wrapper: Currently there are only two types, program normal and program repair
+init_busy            | output | The flash wrapper is undergoing initialization
+tck                  | input  | jtag tck
+tdi                  | input  | jtag tdi
+tms                  | input  | jtag tms
+tdo                  | output | jtag tdo
```

Comment:
we have a JTAG interface struct type now available in `jtag_pkg.sv` if you want to use that.

Reply:
o sounds good. I'll make sure to update to that.